Library/Core/FileResource.swift

```diff
@@ -11,7 +11,7 @@ import Foundation
 public protocol FileResourceType {
```

Comment:
Should all protocols be renamed from `FileResourceType` to `FileResourceProtocol`?
Like how Swift 2's `ErrorType` becomes `ErrorProtocol` in Swift 3.